osbs/build/plugins_configuration.py

```diff
@@ -0,0 +1,583 @@
+
+    def render_inject_parent_image(self):
+        phase = 'prebuild_plugins'
+        plugin = 'inject_parent_image'
+        if not self.template_has_plugin_conf(phase, plugin):
+            return
+
+        koji_parent_build = self.user_params.koji_parent_build.value
+
+        if not koji_parent_build:
+            logger.info('removing %s, koji_parent_build must be provided', plugin)
+            self.remove_plugin(phase, plugin)
+            return
+
+        self.template_set_arg(phase, plugin, 'koji_parent_build', koji_parent_build)
+
+    def render_koji_promote(self, use_auth=None):
+        if not self.template_has_plugin_conf('exit_plugins', 'koji_promote'):
+            return
+
+        koji_target = self.user_params.koji_target.value
+        if not self.set_plugin_arg_valid('exit_plugins', 'koji_promote', 'target', koji_target):
+            logger.info("removing koji_promote from request as no kojihub "
+                        "user_paramsified")
+            self.remove_plugin("exit_plugins", "koji_promote")
+
+    def render_koji_upload(self, use_auth=None):
+        phase = 'postbuild_plugins'
+        name = 'koji_upload'
+        if not self.template_has_plugin_conf(phase, name):
+            return
+
+        def set_arg(arg, value):
+            self.template_set_arg(phase, name, arg, value)
+
+        set_arg('build_json_dir', self.user_params.build_json_dir.value)
+        set_arg('platform', self.user_params.platform.value)
+        set_arg('report_multiple_digests', True)
+
+    def render_koji_import(self, use_auth=None):
+        if not self.template_has_plugin_conf('exit_plugins', 'koji_import'):
+            return
+
+        koji_target = self.user_params.koji_target.value
+        if not self.set_plugin_arg_valid('exit_plugins', 'koji_import', 'target', koji_target):
+            logger.info("removing koji_import from request as no kojihub user_paramsified")
+            self.remove_plugin("exit_plugins", "koji_import")
+
+    def render_koji_tag_build(self):
+        phase = 'exit_plugins'
+        plugin = 'koji_tag_build'
+        if not self.template_has_plugin_conf(phase, plugin):
+            return
+
+        if not self.user_params.koji_target.value:
+            logger.info('Removing %s because no koji_target was user_paramsified', plugin)
+            self.remove_plugin(phase, plugin)
+            return
+
+        self.template_set_arg(phase, plugin, 'target', self.user_params.koji_target.value)
+
+    def render_orchestrate_build(self):
+        phase = 'buildstep_plugins'
+        plugin = 'orchestrate_build'
+        if not self.template_has_plugin_conf(phase, plugin):
+            return
+
+        if self.user_params.platforms.value is None:
+            logger.debug('removing %s plugin: no platforms', plugin)
+            self.remove_plugin(phase, plugin)
+            return
+
+        # Parameters to be used in call to create_worker_build
+        build_kwargs = {
```

Comment:
Should this be using a variation of `self.user_params.to_json()` that returns a `dict` instead?

Reply:
It's identical to the code from BuildRequest, so I think we only want that subset of build parameters.

If you think we should pass all the user parameters, it would be easy if not efficient to do build_kwargs = json.loads(self.user_params.to_json())

Reply:
You could add `to_dict()` helper method in `BuildUserParams`, which can also be used by `to_json`.

Reply:
Fixed.